list.h

```diff
@@ -1,97 +1,43 @@
- *       data members in order to 1) make it obvious that they are private data
- *       members, and 2) avoid name conflicts with functions (e.g. `mLength`
- *       could not be called simply `length` because `length()` is a function,
- *       and `length` is short for `&length`, which is the address of the
- *       function `length()`) and other data members.
- *
- * - We use `length()` instead of `getLength()` in keeping with the `length()`
- *   method on `string`, and because it very well could be more than just a
- *   getter: if we the implementation doesn't have an `mLength` variable, or
- *   similar, `length()` will have to count the elements in the list.
- *
- * - What exactly a `Node` is will vary depending on the implementation, so
- *   each implementation must create its own.
- *     - For example, a singly linked list might have `Node` defined as
- *       `private` inside its class, as a `struct` containing a `Node * next`
- *       and an `int value`.  A doubly linked list would also need its `Node`
- *       to contain a `Node * previous`.
- */
-
 #ifndef LIST_H
 #define LIST_H
-// ----------------------------------------------------------------------------
 
-class List {
-    public:
+//pure virtual class
+class Linkedlist{
+	virtual void addNode(int) = 0;
+	virtual void deleteNode(int) = 0;
+	virtual void printList(int) = 0;
+	virtual void calcLength(int) = 0;
 
-        virtual ~List() = default;
-        /**
-         * Destruct a list, first destructing all elements of the list.
-         */
+	
+};
 
-        virtual int length() const = 0;
-        /**
-         * Returns the number of elements in the list.
-         */
+class List{
 
-        virtual void insert(int index, const int value) = 0;
-        /**
-         * Insert the value in the list so that its `Node` has index `index`
-         * (normalized as described above).
-         */
+public:
 
-        virtual int remove(int index) = 0;
-        /**
-         * Remove the `Node` at `index` (normalized as described above) from
-         * the list, and return the value it contained (or `0` if the list is
-         * empty).
-         */
+	List();
 
-        virtual int peek(int index) const = 0;
-        /**
-         * Return the value inside the `Node` at `index` (normalized as
-         * described above) (or return `0` if the list is empty), without
-         * modifying the list.
-         */
-};
+	void addNode(int,int);
+	void deleteNode(int,int);
+	void printList();
+	void calcLength();
+
+	~List(){};
 
-// ----------------------------------------------------------------------------
-#endif  // LIST_H
+private:
+
+	struct Node{
+		int data;
+		Node * next;
+	};
+
+	typedef struct Node * nodeptr;
+
+	nodeptr node;		//use to create new node
+	nodeptr head;		//always point to the first node of the list
+	nodeptr curr1;		//point to the node before the interest node
+	nodeptr curr2;		//point to the interest node
+	int option = 1;
```

Comment:
`node`, `curr1` and `curr2` could probably be local to the functions in which they are used, rather than global to an instance of a `List`